opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/rxjava/RxJavaTracingAutoConfiguration.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2017-2018 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.spring.cloud.rxjava;
+
+import io.opentracing.Tracer;
+import io.opentracing.rxjava.TracingRxJavaUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Configuration;
+
+
+@Configuration
+@ConditionalOnBean(Tracer.class)
+@ConditionalOnClass(name = "rx.plugins.RxJavaHooks")
+@ConditionalOnProperty(name = "opentracing.spring.cloud.rxjava.enabled", havingValue = "true", matchIfMissing = true)
+public class RxJavaTracingAutoConfiguration {
+
+  @Autowired
+  public RxJavaTracingAutoConfiguration(Tracer tracer) {
```

Comment:
@geoand this looks similar to our recent discussion https://github.com/opentracing-contrib/java-spring-cloud/pull/97#discussion_r168778689